source/Halibut/Transport/TcpClientExtensions.cs

```diff
@@ -43,6 +43,7 @@ void DisposeClient()
             {
                 try
                 {
+                    client.CloseImmediately();
```

Comment:
Just wondering, what was the motivation for this?

Reply:
I was having a whole lot of trouble making things cancel to the socket and thought something might be wrong in async Halibut so started trying to ensure everything was closed and disposed.  Turned out to be a pretty stupid mistake in my test but I did add this, which seemed right and decided to leave it. 

I'm happy to revert from this PR, as you said it's unrelated, and we can add in another PR to remove multiple concerns being changed.

Reply:
Done